app/components/workflow-toggle.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import Translate from 'react-translate-component';
+
+class WorkflowToggle extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleToggleActive = this.handleToggleActive.bind(this);
+    this.state = {
+      error: null,
+      setting: {}
+    };
+  }
+
+  handleToggleActive(property, value) {
+    const setting = this.state.setting;
+    setting[property] = true;
+    this.setState({ error: null, setting });
+
+    const changes = {};
+    changes[property] = value;
+
+    this.props.workflow.update(changes).save()
+      .catch(error => this.setState({ error }))
+      .then(() => {
+        const changedSetting = this.state.setting;
+        changedSetting[property] = false;
+        this.setState({ setting: changedSetting });
+      });
+  }
+
+  render() {
+    const workflow = this.props.workflow;
+    const setting = workflow[this.props.field];
+
+    return (
+      <span>
+        { this.props.workflow.id } - { this.props.workflow.display_name}:
+        <label>
+          <input
+            type="checkbox"
+            name={this.props.field}
+            value={setting}
```

Comment:
The value is being set from a property from the workflow prop. The input onChange handler is also setting it in the state, but `this.state.setting` isn't being used anywhere.

There are two ways to handle inputs in React: either make them controlled or uncontrolled or in other words either have React handle the state of the input or have the native DOM behavior handle the state of the input. There's sort of a mix of both ways going on here. [Read more about controlled inputs](https://reactjs.org/docs/forms.html). We need to pick one or the other, otherwise, bugs can arise because we're not sure which state is accurate, the workflow prop vs the local component state.

I would suggest simplifying the state a bit here too. We don't want to exactly replicate what the `SetToggle` mixin did. Mixins were a precursor to higher order components, so this mixin was in the past used in several components and could be managing the state of several toggles. We're removing it now, so we're not really managing multiple settings at a time anymore. The component state could just be the current value of the checkbox, if you decide to go the route of a controlled input.

There also has to be consideration of not being out of sync of the state of the workflow too.

Let me know when you've read up on this and we discuss about how to solve all of these state management issues.